src/iot/fieldGroupings.spec.js

```diff
@@ -0,0 +1,583 @@
+          expectedGrouping
+        );
+      });
+    });
+
+    context('when there is missing required information', function() {
+      it('throws an error when facilityId is missing', function() {
+        const fieldGrouping = fixture.build('fieldGrouping');
+        const fieldGroupings = new FieldGroupings(
+          baseSdk,
+          baseRequest,
+          expectedHost
+        );
+        const promise = fieldGroupings.create(null, fieldGrouping);
+
+        return expect(promise).to.be.rejectedWith(
+          'A facilityId is required for getting information about a field grouping.'
+        );
+      });
+
+      ['label', 'description'].forEach(function(field) {
+        it(`throws an error when ${field} is missing`, function() {
+          const facilityId = fixture.build('facility').id;
+          const fieldGrouping = fixture.build('fieldGrouping');
+          const fieldGroupings = new FieldGroupings(
+            baseSdk,
+            baseRequest,
+            expectedHost
+          );
+          const promise = fieldGroupings.create(
+            facilityId,
+            omit(fieldGrouping, [field])
+          );
+
+          return expect(promise).to.be.rejectedWith(
+            `A ${field} is required to create a new field grouping.`
+          );
+        });
+      });
+    });
+  });
+
+  describe('delete', function() {
+    context('the field grouping ID is provided', function() {
+      let fieldGroupingId;
+      let promise;
+      let request;
+
+      beforeEach(function() {
+        fieldGroupingId = fixture.build('fieldGrouping').id;
+
+        request = {
+          ...baseRequest,
+          delete: this.sandbox.stub().resolves()
+        };
+
+        const fieldGroupings = new FieldGroupings(baseSdk, request);
+        fieldGroupings._baseUrl = expectedHost;
+
+        promise = fieldGroupings.delete(fieldGroupingId);
+      });
+
+      it('deletes the field grouping from the server', function() {
+        expect(request.delete).to.be.calledWith(
+          `${expectedHost}/groupings/${fieldGroupingId}`
+        );
+      });
+
+      it('returns a fulfilled promise', function() {
+        return expect(promise).to.be.fulfilled;
+      });
+    });
+
+    context('the field grouping ID is not provided', function() {
+      it('throws an error', function() {
+        const fieldGroupings = new FieldGroupings(baseSdk, baseRequest);
+        const promise = fieldGroupings.delete();
+
+        return expect(promise).to.be.rejectedWith(
+          'A groupingId is required for deleting a field grouping.'
+        );
+      });
+    });
+  });
+
+  describe('get', function() {
+    context('the field grouping ID is provided', function() {
+      let expectedFieldGrouping;
+      let promise;
+      let rawFieldGrouping;
+      let request;
+      let toCamelCase;
+
+      beforeEach(function() {
+        expectedFieldGrouping = fixture.build('fieldGrouping');
+        rawFieldGrouping = fixture.build(
```

Comment:
Can you change this to match the vernacular used for other objects going to and from the API server?